### PR TITLE
Issue#52

### LINKED ISSUE #52: Deal with EOL Whitespace in outputted JSON
According to the [`json.dump` docs](https://docs.python.org/3.5/library/json.html#json.dump), the default `separators` value for Python 2 is `(', ', ': ')`, even with a non-default `indent` value.

This means before the `\n` at the end of lines with terminal commas there is a space. EOL Whitespace! Sound the alarm!

This was fixed in Python v3.4, but as the docs recommend and the current production job use `python` instead of `python3` this means we need to specify `separators=(',',': ')` in `json.dump`

### PR DESCRIPTION
**Pull request by jhabarsingh**
Fixes #52

### CODE REVIEW

**Comment by jhabarsingh** [comment]
okay sir, can you tell me the file name where I have to write tests .

**Comment by jhabarsingh** [comment]
```
{
   "test1": 1,
   "test2": 2,
   "test3": 3
}
```
this is the output it will give.
it's giving space after colon, is that okay or not ?

I wanted to confirm it once before writing the tests.

**Comment by mdboom** [comment]
You can put the test here https://github.com/mozilla/probe-scraper/blob/master/tests/test_runner.py

**Comment by jhabarsingh** [comment]
> ```
> {
>    "test1": 1,
>    "test2": 2,
>    "test3": 3
> }
> ```
> 
> this is the output it will give.
> it's giving space after colon, is that okay or not ?
> 
> I wanted to confirm it once before writing the tests.

**Comment by jhabarsingh** [comment]
Sir I have added tests to check trailing spaces in the output of  json.dump(), can you review it and let me know if any changes needed

File: tests/test_runner.py [hunk 1]
```diff
@@ -69,3 +71,35 @@ def test_add_first_appeared_dates():
     }
 
     assert runner.add_first_appeared_dates(probes_by_channel, first_appeared_dates) == expected
+
+
+def test_trailing_space():
```
Comment (mdboom) [inline tests/test_runner.py:76]: This test dumps files in the current directory, which is generally not a good idea.  Let's use the [tmp_path](https://docs.pytest.org/en/stable/tmpdir.html#the-tmp-path-fixture) fixture here instead.

File: tests/test_runner.py [hunk 2]
```diff
@@ -69,3 +71,35 @@ def test_add_first_appeared_dates():
...
+    '''
+    test_cases = [
+        {
+            "test1": 12,
+            "test2": 31,
+        },
+        {
+            "test1": 12,
+            "test2": 31
+        },
```
Comment (mdboom) [inline tests/test_runner.py:87]: Both of these test cases are the same.  I think it's fine to just have one test case.

Comment (jhabarsingh) [reply tests/test_runner.py:87]: yeah I will remove this portion

File: tests/test_runner.py (hunk 1 above)
Comment (jhabarsingh) [reply tests/test_runner.py]: Okay sir I will use tmp_path intead

**Comment by jhabarsingh** [comment]
@mdboom,  I have had the changes you told me can you review it once ?

File: tests/test_runner.py [hunk 3]
```diff
@@ -69,3 +71,31 @@ def test_add_first_appeared_dates():
...
+        path = os.path.join(DIR_NAME, FILE_NAME)
+        with open(path, 'r') as file:
+            for line in file.readline():
+                if line[-1] == ' ':
+                    trailing_spaces += 1
+
+        assert not trailing_spaces
+
+    # Deleting the files created [ TEAR-DOWN ]
+    shutil.rmtree(DIR_NAME)
```
Comment (mdboom) [inline tests/test_runner.py:101]: This isn't necessary if using the `tmp_path` fixture.

**Comment by jhabarsingh** [comment]
I have removed the  lines that were using shutil.

**Comment by jhabarsingh** [comment]
Sir, this was my first contribution to the Open source and I am glad you were there to help me. Thanks a lot for being so helpful.

**Pull request merged**